### PR TITLE
MAINT: simplify code that assumes str/unicode and int/long are different types

### PR DESCRIPTION
**Pull request by eric-wieser**
Cleanup from the dropping of python 2

### CODE REVIEW

**Comment by mattip** [comment]
At some point we should convert uses of np.random to the new API

File: numpy/core/tests/test_multiarray.py [hunk 1]
```diff
@@ -1512,7 +1512,7 @@ def test_dumps(self):
 
     def test_pickle(self):
         for proto in range(2, pickle.HIGHEST_PROTOCOL + 1):
-            for dt in [bytes, np.void, unicode]:
+            for dt in [bytes, np.void, str]:
                 zs = self._zeros(10, dt)
                 p = pickle.dumps(zs, protocol=proto)
                 zs2 = pickle.loads(p)
```
Comment (seberg) [inline numpy/core/tests/test_multiarray.py]: Below in line 4459 there is another one of these (causes the test failures). Probably I will just fix it up myself and merge later.

**Review by anirudh2290 — COMMENTED** [review]
I see `numpy/compat/py3k.py` unicode = str, long = int. Is that needed anymore ?
Also, i see unicode usage in `numpy/core/_type_aliases.py` does that also need to be cleaned up ?

**Comment by eric-wieser** [comment]
> I see `numpy/compat/py3k.py` unicode = str, long = int. Is that needed anymore ?

Sadly yes, `np.unicode` and `np.long` come from here and are public API until someone writes an NEP for #14882.

> Also, i see unicode usage in numpy/core/_type_aliases.py does that also need to be cleaned up ?

Probably, but that can always be a follow-up.

**Comment by seberg** [comment]
Add the (hopefully last) missing cases, and will merge later once the tests succeed.

**Comment by mattip** [comment]
doc build fails: `WARNING: [autosummary] failed to import \
'numpy.ma.MaskedArray.__long__': no module named numpy.ma.MaskedArray.__long__`

**Comment by seberg** [comment]
Thanks for the ping, removed it from the docs, lets see if it will finally pass...

**Comment by seberg** [comment]
Thanks Eric and Matti, seems this is ready to go in after deleting that line from the documentation.

**Pull request merged**